aea/cli/utils/click_utils.py

```diff
@@ -0,0 +1,119 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""Module with click utils of the aea cli."""
```

Comment:
I feel like the classes in this file would better belong in other parts:
- `ConnectionsOption` with any other code about option parsing
- `PublicIdParameter` and `AgentDirectory` with other code about context handling

Reply:
It looks like that. But for the purpose of usage (expanding the click commands functionality with custom things) all of them playing a similar role. We don't use it directly for parsing or validating parameter, `click` tool uses.